odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,124 @@ def _validate_module_views(self, module):
                 view._check_xml()
             except Exception as e:
                 self.raise_view_error("Can't validate view:\n%s" % e, view.id)
+
+    def _fix_faulty_views(self, delete=False):
+        """ Use the _get_faulty_views and take appropriate actions to fix the issue """
+        for view in self:
+            view_id = view.id # allow to use even if view is deleted
+            faulty_views = view._get_faulty_views()
+            if faulty_views:
+                self._cr.execute('SAVEPOINT end_validation')
+                faulty_views._clean_properly(delete)
+                self._cr.execute("select count(id) from ir_ui_view where id = %s", (view_id,))
+                cnt = self.env.cr.fetchone()[0]
+                if cnt == 0 or view._is_valid():
+                    _logger.info("View %s can be fixed by disabling/deleting following views: %s" % (view_id,faulty_views))
+                    self._cr.execute('RELEASE end_validation')
+                else:
+                    self._cr.execute('ROLLBACK TO end_validation')
+                    self._cr.execute('RELEASE end_validation')
+                    _logger.warning("Issue occurs: The proposition made to fix the view %s (disable/delete %s) is not recommended" % (view,faulty_views))
+                    # should not happend
+
+    def _get_faulty_views(self,parent_view=None):
+        """ Returns the faulty views
+            To determine the faulty views, we travel the tree recursively.
+            Disable all children, then enable one by one. If there is an issue.
+            The activated node or at least one of its children are faulty.
+            Disable faulty node and continue recursion.
+            This function is never called from odoo code but allows to be called from a server action.
+            Please don't remove
+    usage:
+        view1._get_faulty_views()
+        """
+        self.ensure_one()
+        disabled_views = self.env['ir.ui.view']
+        parent_view = parent_view or self
+        if parent_view==self:
+            if self._is_valid():
+                return disabled_views
+            self._cr.execute('SAVEPOINT test_fix_views')
+            temp_disabled_views=self._deactivate_lvl1_children() #ordered by priority ( by default)
+            if self._is_valid():
+                if len(self.inherit_children_ids) or len(temp_disabled_views)>1:
+                    for child_view in (temp_disabled_views):
+                        for v in child_view._get_faulty_views(self):
+                            disabled_views|=v       
+                else:
+                    self.active=False
+                    disabled_views+=self
```

Comment:
let's use set notation consistently through the whole method if you don't mind :)
`disabled_views|=self`

Reply:
done